### PR TITLE
Update Ethash cache implementation

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/pop/crypto/progpow.hpp [hunk 1]
```diff
@@ -6,9 +6,8 @@
 #ifndef VERIBLOCK_POP_CPP_PROGPOW_HPP
 #define VERIBLOCK_POP_CPP_PROGPOW_HPP
 
+#include <veriblock/pop/crypto/progpow/cache.hpp>
```
Comment (Warchant) [inline include/veriblock/pop/crypto/progpow.hpp:9]: for local includes use #include "file.h"

File: include/veriblock/pop/algorithm.hpp [hunk 2]
```diff
@@ -132,7 +130,7 @@ void erase_if(container_t& c, std::function<bool(const val_t&)> pred) {
 //! @private
 template <typename T, typename... Args>
 std::unique_ptr<T> make_unique(Args&&... args) {
-  auto* ptr = new T(std::forward<Args...>(args)...);
+  auto* ptr = new T(args...);
```
Comment (Warchant) [inline include/veriblock/pop/algorithm.hpp:133]: why?

File: src/pop/crypto/progpow/progpow.cpp [hunk 3]
```diff
@@ -2,6 +2,11 @@
 // https://www.veriblock.org
 // Distributed under the MIT software license, see the accompanying
 // file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+#include <cstdint>
+#include <mutex>
+#include <utility>
+#include <vector>
+#include <veriblock/pop/algorithm.hpp>
```
Comment (Warchant) [inline src/pop/crypto/progpow/progpow.cpp:9]: use #include "file.h" for project includes

File: test/pop/crypto/progpow_cache_test.cpp [hunk 4]
```diff
@@ -5,11 +5,11 @@
 
 #include <gtest/gtest.h>
 
-#include <veriblock/pop/entities/vbkblock.hpp>
 #include <veriblock/pop/blockchain/alt_chain_params.hpp>
-#include <veriblock/pop/blockchain/vbk_chain_params.hpp>
 #include <veriblock/pop/blockchain/btc_chain_params.hpp>
+#include <veriblock/pop/blockchain/vbk_chain_params.hpp>
 #include <veriblock/pop/crypto/progpow.hpp>
+#include <veriblock/pop/entities/vbkblock.hpp>
```
Comment (Warchant) [inline test/pop/crypto/progpow_cache_test.cpp:12]: use "" includes

**Review by Warchant — CHANGES_REQUESTED** [review]
also rebase on master

File: test/pop/crypto/progpow_cache_test.cpp (hunk 4 above)
Comment (Mr-Leshiy) [reply test/pop/crypto/progpow_cache_test.cpp]: fixed

File: src/pop/crypto/progpow/progpow.cpp (hunk 3 above)
Comment (Mr-Leshiy) [reply src/pop/crypto/progpow/progpow.cpp]: fixed

File: include/veriblock/pop/algorithm.hpp (hunk 2 above)
Comment (Mr-Leshiy) [reply include/veriblock/pop/algorithm.hpp]: fixed

File: include/veriblock/pop/crypto/progpow.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply include/veriblock/pop/crypto/progpow.hpp]: fixed

**Review by Warchant — APPROVED** [review]

**Pull request merged**